launchable/test_runners/cucumber.py

```diff
@@ -304,51 +305,95 @@ def _create_file_candidate_list(file: str) -> List[str]:
     return list
 
 
-class HookTestCaseInfo:
-    def __init__(self, duration: int, statuses: List[str], stderr: List[str]) -> None:
-        self.statuses = statuses
-        self.stderr = stderr
-        self.duration = duration  # nano sec
+class Result:
+    def __init__(self, statuses: List[str], duration_nano_sec: int, error_message: List[str]) -> None:
+        self._statuses = statuses
+        self._duration_nano_sec = duration_nano_sec
+        self._error_message = error_message
 
 
-def _extract_test_case_info_from_hook(data):
-    duration = 0  # nano sec
+class TestCaseHookInfo(Result):
+    def __init__(self, duration_nano_sec: int, statuses: List[str], stderr: List[str]) -> None:
+        super().__init__(statuses=statuses, duration_nano_sec=duration_nano_sec, error_message=stderr)
+
+
+def _parse_hook_from_element(element: Dict[str, List]) -> TestCaseHookInfo:
```

Comment:
This might be too magical, so it's up to you. However, we can write like this.
```py
def _parse_hook_from_element(element: Dict[str, List]) -> TestCaseHookInfo:
    duration_nano_sec = 0
    statuses = []
    stderr = []
    def _parse_step_result(step: Dict[str, Dict]):
        result = step.get("result", None)
        if result:
            nonlocal duration_nano_sec
            duration_nano_sec += result.get("duration", 0)
            statuses.append(result.get("status"))
            if result.get("error_message", None):
                stderr.append(result["error_message"])
    for step in element.get("before", []):
        _parse_step_result(step)
    for step in element.get("after", []):
        _parse_step_result(step)

    return TestCaseHookInfo(
        duration_nano_sec=duration_nano_sec,
        statuses=statuses,
        stderr=stderr
    )
```

Reply:
Thanks I apply it